src/Fixer/Casing/DeclaredClassCasingFixer.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\Casing;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\Tokenizer\CT;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @author siad007
+ */
+final class DeclaredClassCasingFixer extends AbstractFixer
+{
+    private static $declaredClassNames;
+
+    public function __construct()
+    {
+        parent::__construct();
+
+        if (null === self::$declaredClassNames) {
+            foreach (get_declared_classes() as $class) {
+                self::$declaredClassNames[strtolower($class)] = $class;
+            }
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'Classes should be referred to using the correct casing.',
+            [new CodeSample("<?php\nnew STDCLASS();\n")]
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return $tokens->isTokenKindFound(T_STRING);
+    }
+
+    /**
+     * @param \SplFileInfo $file
+     * @param Tokens       $tokens
+     */
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
+    {
+        foreach ($tokens as $index => $token) {
+            if ($name = $this->declaredClass($index, $token, $tokens)) {
+                $tokens[$index] = new Token([T_STRING, self::$declaredClassNames[$name]]);
+            }
+        }
+    }
+
+    /**
+     * Get the lower case name of the declared class or null.
+     *
+     * @param int    $index
+     * @param Token  $token
+     * @param Tokens $tokens
+     *
+     * @return null|string
+     */
+    private function declaredClass($index, Token $token, Tokens $tokens)
+    {
+        $beforeClassName = $tokens->getPrevMeaningfulToken($index);
+        $lower = strtolower($token->getContent());
+
+        $isDeclaredClass = null;
```

Comment:
When I see a method or variable starting with `is` I assume it's a boolean, I think this variable name could be clearer.